Mathlib/Data/Matroid/Basic.lean

```diff
@@ -597,6 +596,19 @@ theorem Indep.exists_insert_of_not_base (hI : M.Indep I) (hI' : ¬M.Base I) (hB
   exact ⟨e, ⟨he.1, not_mem_subset hIB' he.2⟩,
     ⟨_, hBase, insert_subset_insert (subset_diff_singleton hIB' hx)⟩⟩
 
+/-- This is the same as `Indep.exists_insert_of_not_base`, but phrased so the statement is
+  defeq to the augmentation axiom for independent sets -/
+theorem aug_property (M : Matroid α) :
+    ∀⦃I B⦄, M.Indep I → I ∉ maximals (· ⊆ ·) (setOf M.Indep) →
+      B ∈ maximals (· ⊆ ·) (setOf M.Indep) → ∃ x ∈ B \ I, M.Indep (insert x I) := by
+  intro I B hI hImax hB
```

Comment:
Doing the intros separately makes it harder to see what the names refer to:
```suggestion
theorem aug_property (M : Matroid α) ⦃I B⦄ (hI : M.Indep I) (hImax : I ∉ maximals (· ⊆ ·) (setOf M.Indep))
      (hB : B ∈ maximals (· ⊆ ·) (setOf M.Indep)) : ∃ x ∈ B \ I, M.Indep (insert x I) := by
```

Reply:
As per the dosctring, this lemma is designed to be plugged directly into a construction of another matroid, rather than invoked with explicit I and B. (See how it is invoked in the next file). The version for use with a particular `I` and `B` is `Indep.exists_insert_of_not_base`, which is written in the normal way with everything before the colon.

Reply:
Moving things before the colon is unrelated to whether arguments are taken explicitly or implicitly.

Reply:
Ok - changed.